common/src/main/java/bisq/common/crypto/Encryption.java

```diff
@@ -230,6 +232,24 @@ public static SecretKey generateSecretKey(int bits) {
         }
     }
 
+    /**
+     * Generates a secret key from password string.
+     * 
+     * @param p Password
+     * @param s Salt
+     * @param i Iterations
+     * @param l Key length
+     * @return
+     * @throws Exception
+     */
+    public static SecretKey generateSecretKey(String p, byte[] s, int i, int l) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        PBEKeySpec ks = new PBEKeySpec(p.toCharArray(), s, i, l);
+        SecretKeyFactory skf = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA512");
+        SecretKey secret = skf.generateSecret(ks);
+        SecretKeySpec secretKeySpec = new SecretKeySpec(secret.getEncoded(), Encryption.SYM_KEY_ALGO);
```

Comment:
what does this line do? the `SecretKey secret` from the previous line cannot be returned?

Reply:
No it cannot be returned directly, the SecretKeySpec must have AES specified. The function naming could be clarified that it is generating an AES secret key.

Reply:
Deleted